Ruby_Tutorial/bst.rb

```diff
@@ -0,0 +1,228 @@
+class BinarySearchTree
+	attr_accessor :root, :data
+		
+	def initialize value
+		@root = nil
+	end
+
+	def insert(value)
+		new_node = Node.new(value)
+
+		if !root
+			@root = new_node
+			return self
+		end
+
+		current = @root
+		while current do
+			return nil if current.value == value
+			if (value > current.value)
+				if !current.right
+					current.right = new_node
+					break
+				else
+					current = current.right
+				end
+			else
+				if !current.left
+					current.left = new_node
+					break
+				else
+					current = current.left
+				end
+			end
+		end
+		return self
+	end
+
+	def find(value, current = @root)
+		return false if !current
+		return true if current.value == value
+		return find(value, current.left) if value < current.value
+		return find(value, current.right) if value > current.value
+	end
+
+	def largest(current = @root)
+		while current do
+			return current if current.right == nil
+			current = current.right
+		end
+	end
+
+	def smallest(current = @root)
+		while current do
+			return current if current.left == nil
+			current = current.left
+		end
+	end
+	
+	def preorder(current = @root)
+		return if (current == nil)
+		puts current.value.to_s
+		preorder(current.left)
+		preorder(current.right)
+	end
+
+	def postorder(current = @root)
+		return if (current == nil)
+		postorder(current.left)
+		postorder(current.right)
+		puts current.value.to_s
+	end
+
+	def inorder(current = @root)
+		return if (current == nil)
+		inorder(current.left)
+		puts current.value.to_s
+		inorder(current.right)
+	end
+
+	def level_order(current = @root)
+		return if (current == nil)
+		puts current.value.to_s
+		level_order(current.left)
+		level_order(current.right)
+	end
+
+	def data
+		@data ||= []
+	end
+
+	def store_elements(current = @root)
```

Comment:
use order formats

Reply:
Order methods are build to print the output on terminal, so to avoid that its better to have different storing method